x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/select_rule_type/index.test.tsx

```diff
@@ -186,7 +186,7 @@ describe('SelectRuleType', () => {
     });
 
```

Comment:
```suggestion

    it('renders selected card only when in update mode for "esql" and esql feature is disabled', () => {
      useEsqlAvailabilityMock.mockReturnValueOnce(false);
      const field = useFormFieldMock<unknown>({ value: 'esql' });
      const wrapper = mount(
        <TestProviders>
          <SelectRuleType
            describedByIds={[]}
            field={field}
            isUpdateView={true}
            hasValidLicense={true}
            isMlAdmin={true}
          />
        </TestProviders>
      );
      expect(wrapper.find('[data-test-subj="customRuleType"]').exists()).toBeFalsy();
      expect(wrapper.find('[data-test-subj="machineLearningRuleType"]').exists()).toBeFalsy();
      expect(wrapper.find('[data-test-subj="thresholdRuleType"]').exists()).toBeFalsy();
      expect(wrapper.find('[data-test-subj="eqlRuleType"]').exists()).toBeFalsy();
      expect(wrapper.find('[data-test-subj="threatMatchRuleType"]').exists()).toBeFalsy();
      expect(wrapper.find('[data-test-subj="esqlRuleType"]').exists()).toBeTruthy();
    });

```


Let's also add unit test for the case covered in https://github.com/elastic/kibana/pull/181616/files#r1585195094

Reply:
Done, thanks!